shared/macros/10-ansible.jinja

```diff
@@ -1022,3 +1049,361 @@ Part of the grub2_bootloader_argument_absent template.
+    - result_pam_line_present.found == 0
+{{%- endmacro -%}}
+
+
+{{#
+    Make sure that an existing PAM module line is properly configured with an option.
+
+:param pam_file:        PAM config file.
+:param group:           PAM management group: auth, account, password or session. Also known as "type".
+:param control:         PAM control flags.
+:param module:          PAM module name.
+:param option:          PAM module option.
+:param value:           PAM module option argument, if is case. Optional parameter.
+:param after_match:     Regex used as reference to include the PAM line below, if necessary. Optional parameter.
+
+#}}
+{{%- macro ansible_ensure_pam_module_option(pam_file, group, control, module, option, value='', after_match='') -%}}
+{{{ ansible_ensure_pam_module_line(pam_file, group, control, module, after_match) }}}
+
+- name: '{{{ rule_title }}} - Check if the required PAM module option is present in {{{ pam_file }}}'
+  ansible.builtin.lineinfile:
+    path: "{{{ pam_file }}}"
+    regexp: ^\s*{{{ group }}}\s+{{{ control }}}\s+{{{ module }}}\s*.*\s{{{ option }}}\b
+    state: absent
+  check_mode: true
+  changed_when: false
+  register: result_pam_module_{{{ option }}}_option_present
+
+- name: '{{{ rule_title }}} - Ensure the "{{{ option }}}" PAM option for "{{{ module }}}" is included in {{{ pam_file }}}'
+  ansible.builtin.lineinfile:
+    path: "{{{ pam_file }}}"
+    backrefs: true
+    regexp: ^(\s*{{{ group }}}\s+{{{ control }}}\s+{{{ module }}}.*)
+    {{%- if value == '' %}}
+    line: \1 {{{ option }}}
+    {{%- else %}}
+    line: \1 {{{ option }}}={{{ value }}}
+    {{%- endif %}}
+    state: present
+  register: result_pam_{{{ option }}}_add
+  when:
+    - result_pam_module_{{{ option }}}_option_present.found == 0
+
+{{%- if value != '' %}}
+- name: '{{{ rule_title }}} - Ensure the required value for "{{{ option }}}" PAM option from "{{{ module }}}" in {{{ pam_file }}}'
+  ansible.builtin.lineinfile:
+    path: "{{{ pam_file }}}"
+    backrefs: true
+    regexp: ^(\s*{{{ group }}}\s+{{{ control }}}\s+{{{ module }}}\s+.*)({{{ option }}})=[0-9a-zA-Z]+\s*(.*)
+    line: \1\2={{{ value }}} \3
+  register: result_pam_{{{ option }}}_edit
+  when:
+    - result_pam_module_{{{ option }}}_option_present.found > 0
+{{%- endif %}}
+{{%- endmacro -%}}
+
+
+{{#
+    Remove a PAM module option if present in a PAM module line.
+
+:param pam_file:        PAM config file.
+:param group:           PAM management group: auth, account, password or session. Also known as "type".
+:param control:         PAM control flags. Optional parameter, but recommended to be informed whenever possible.
+:param module:          PAM module name.
+:param option:          PAM module option.
+
+#}}
+{{%- macro ansible_remove_pam_module_option(pam_file, group, control, module, option) -%}}
+- name: '{{{ rule_title }}} - Ensure the "{{{ option }}}" option from "{{{ module }}}" is not present in {{{ pam_file }}}'
```

Comment:
This task also needs to be run 3 times to converge:
`ansible-playbook -i "rhel87n," --tags display_login_attempts build/ansible/rhel8-playbook-stig.yml`

I have no idea why it doesn't coverge on the second run.

Reply:
Answered here https://github.com/ComplianceAsCode/content/pull/9097#discussion_r916571769